Tasks/VSMobileCenterUpload/utils.ts

```diff
@@ -119,16 +139,66 @@ export function resolveSinglePath(pattern: string, continueOnError: boolean): st
             }
         }
 
-        if (matches.length != 1) {
-            if (continueOnError) {
-                tl.warning(tl.loc("FoundMultipleFiles", pattern));
-            } else {
-                throw new Error(tl.loc("FoundMultipleFiles", pattern));
-            }
+        return removeDuplicates(matches.map(v => packParentFolder ? path.dirname(v) : v));
+    }
+
+    return null;
+}
+
+export function removeDuplicates(list: string[]): string[] {
+    let unique: string[] = [];
+
+    list.forEach(s => {
```

Comment:
Not tested, not sure if works.  Arrays in JavaScript are associative arrays, I think we can avoid the "find" call if you just push each element in list as a key in the array, and get all unique values from `Object.keys(array)` call.  Again I didn't test this yet.